### PR TITLE
Wire modifications- Unit Tests

### PR DESCRIPTION
**Pull request by M-JULIANI**
### Note
Make sure to merge #23 prior to reviewing this PR.

### Purpose
This separate PR includes the unit testing for the following PR: #23.

### Description
This PR includes unit tests for Wire Behaviour functionality. It is a separate PR for clarity of scope. PR #23 needs to be merged prior to this PR being merged.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@SHKnudsen 

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: test/DynamoCoreWpfTests/RecordedTests.cs [hunk 1]
```diff
@@ -2030,8 +2030,8 @@ public void Defect_MAGN_491_DS()
             ConnectorViewModel secondConnector = connectors[1];
 
             // Find out the corresponding ports they connect to.
-            Point2D firstPoint = firstConnector.ConnectorModel.End.Center;
-            Point2D secondPoint = secondConnector.ConnectorModel.End.Center;
+            Point2D firstPoint = firstConnector.Model.End.Center;
+            Point2D secondPoint = secondConnector.Model.End.Center;
```
Comment (SHKnudsen) [inline test/DynamoCoreWpfTests/RecordedTests.cs:2034]: how come these have changed?

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            DynamoSelection.Instance.Selection.Add(connectorPin);
+
+            ///initial x, y position of pin
+            double initialX = connectorPin.CenterX;
+            double initialY = connectorPin.CenterY;
+
+            var point = new Point2D(initialX, initialY);
+            var operation = DragSelectionCommand.Operation.BeginDrag;
+            var command = new DragSelectionCommand(point, operation);
+            var selected = DynamoSelection.Instance.Selection;
```
Comment (SHKnudsen) [inline test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: dosent seem to be used anywhere

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            ///execute 'begin drag'
+            Model.ExecuteCommand(command);
+
+            operation = DragSelectionCommand.Operation.EndDrag;
+            point.X += 100;
+            point.Y += 100;
+
+            command = new DragSelectionCommand(point, operation);
```
Comment (SHKnudsen) [inline test/DynamoCoreWpfTests/ConnectorViewModelTests.cs:269]: might need some comments on this, not sure i understand why we are executing this command twice

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        /// </summary>
+        [Test]
+        public void CanHideConnector()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+            ///Toggles hide (visibility == off)
+            connectorViewModel.HideConnectorCommand.Execute(null);
+            Assert.IsFalse(connectorViewModel.IsVisible);
+        }
```
Comment (SHKnudsen) [inline test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: i thought we just set it to transparent, didnt we?

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+            connectorViewModel.PanelX = 352.666666;
+            connectorViewModel.PanelY = 305.333333;
+
+            connectorViewModel.PinConnectorCommand.Execute(null);
+
+            ///Request remove pin
+            var firstPin = connectorViewModel.ConnectorPinViewCollection.First();
+            firstPin.OnRequestRemove(firstPin, System.EventArgs.Empty);
+
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 0);
```
Comment (SHKnudsen) [inline test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: in cases like this, it is generally good to assert that something has actually been removed, this test doesn't actually tell us if it was removed as we dont know what ConnectorPinViewCollection.Count was before we did remove

Comment (SHKnudsen) [reply test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: fyi this goes for most of the tests in this class

**Review by SHKnudsen — CHANGES_REQUESTED** [review]

File: test/DynamoCoreWpfTests/RecordedTests.cs (hunk 1 above)
Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/RecordedTests.cs:2034]: This had to do with the resharper issue that went through and renamed all things `model` with `connectorModel`.. just reverted it for clarity.

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs (hunk 4 above)
Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: Correct, however `IsVisible` is the property/flag that indicates to the xaml (datatrigger) that this is so.

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs (hunk 3 above)
Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/ConnectorViewModelTests.cs:269]: Will do, we are calling it once for `begindrag` and a second time for `enddrag`.

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs (hunk 2 above)
Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: Correct, selected doesn't- removed.

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs (hunk 3 above)
Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/ConnectorViewModelTests.cs:269]: Done.

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs (hunk 4 above)
Comment (SHKnudsen) [reply test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: ohhh right, got this confused with the view `IsVisible` property!

File: test/DynamoCoreWpfTests/ConnectorViewModelTests.cs (hunk 5 above)
Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/ConnectorViewModelTests.cs]: Cool, modified all tests in this class to assert more meaningful things :).

File: test/DynamoCoreWpfTests/RecordedTests.cs (hunk 1 above)
Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/RecordedTests.cs:2034]: This change should have been picked up in the main PR. Will do that now.

Comment (M-JULIANI) [reply test/DynamoCoreWpfTests/RecordedTests.cs:2034]: Changed this back in this PR and in the original PR it has been changed correctly to say `model`.

**Pull request merged**